### ISSUE TITLE
[Enhancement] Setup cron that updates Git repo with latest NVT data 

### ISSUE THREAD

**Issue by mrod23** [Secure-Compliance-Solutions-LLC/GVM-Docker#207]
**Is your feature request related to a problem? Please describe.**
If new OpenVas containers are spun up and have the same external IP, NVT syncs will be blocked. 

**Describe the solution you'd like**
A cron will run every 24 or 48 hours that updates Git with the latest NVT data. Containers can pull data from this Git repo, so they don't get blocked by OpenVas. 

**Describe alternatives you've considered**
We could potentially do this with S3 or Azure blob but we would have to store the data in someone's account which cost money. 

**Additional context**

**Comment by mrod23** [comment]
@austinsonger  can I pick this ticket up?

**Comment by mrod23** [comment]
Let me know what you think about the Github idea for the NVT cache

**Comment by austinsonger** [comment]
@mrod23 yeah that idea sounds good too me.

**Comment by Dexus** [comment]
I think it would be a greate idea, but then we need be sure, that it will run from a secure place.
I'm not sure but does GitHub Pro have some options to run a "Pipeline" like scheduled like azure does?
Or we can use the www.travis-ci.org or similar ones that are "rsync and push to git" really easy.
Don't know why this needs a discussion...

**Comment by mrod23** [comment]
I can figure out a mechanism for automation. 

The main thing I wanted to confirm is if it's cool to add to Github vs Azure vs S3 

If someone wants to offer up their Azure or AWS account. I can use that

**Comment by austinsonger** [comment]
Lets just start with Github right now for the time being.

**Comment by Dexus** [comment]
But we will run into a Issue: filesize of the scap data...
github allow only 100.00 MB

**Comment by mrod23** [comment]
Another idea is to spin up a container that caches all of the nvt data. 

And setup the OpenVas container so it can optionally pull from the cache instead of directly from greenbone

**Comment by pixelsquared** [comment]
We have setup [this project](https://github.com/Secure-Compliance-Solutions-LLC/gvm-data-sync) that syncs the data to Backblaze and allows the archive download with Cloudflare at https://vulndata.securecompliance.solutions/file/VulnData/data.tar.xz

**Comment by austinsonger** [comment]
I'm going to close this for now. This can be reopened later, if needed.

**Comment by mrod23** [comment]
Can we keep this opened?

The questions I have with the current sync-all.sh script (https://github.com/Secure-Compliance-Solutions-LLC/GVM-Docker/blob/master/scripts/sync-all.sh) are listed below:
- What should users do if they lose connectivity to vulndata.securitycompliance.solutions and greenbone? I've had connection issues pinging https://vulndata.securecompliance.solutions/file/VulnData/data.tar.xz and reaching Greenbone from China before 
- How we can confirm the cache is updated on a daily basis?

I think it would be a great option to have a user controlled cache so we don't have to rely on outside resources. 

I think the easiest solution is to have an optional  environment variable SYNC_WITH_CACHE that points to a container that runs the code from [here](https://github.com/Secure-Compliance-Solutions-LLC/gvm-data-sync) and syncs data from that container.

I had something setup in the past that did this. But my setup stopped worked after Greenbone made all of their updates in 2020. 

I tried running Openvas with data exclusively pulled from https://vulndata.securecompliance.solutions/file/VulnData/data.tar.xz. I noticed report formats are missing. There maybe other data missing as well. 

@austinsonger  / @pixelsquared  do you know the diff on what data is pulled from the rsync and greenbone-feed-sync commands and what's in https://vulndata.securecompliance.solutions/file/VulnData/data.tar.xz. If we have that diff, we can setup containers to exclusively pull from their own cache if they want instead needing to have to pull from Greenbone on the first sync. 

If you can help me understand the diff. I can take care of setting up the rest and put a PR together.

**Comment by pixelsquared** [comment]
The report formats not working was due to this https://github.com/Secure-Compliance-Solutions-LLC/GVM-Docker/pull/204

The data that our sync script pull should be the same as the normal sync commands. If you find another issue like the report data not working from the sync mirror please open a new issue.

I have no issues with adding a environment variable to control the data download location. 

I think it would make sense to modify the https://github.com/Secure-Compliance-Solutions-LLC/gvm-data-sync repo and build it in to a docker image that can serve the data.tar.xz.

**Comment by Dexus** [comment]
Hey mates, I will build something next week or in the next two weeks. Also I would change to alpine since this image would be only 430mb and not 2.7gb in size.

The Idea with the container based sync is the best. And export it from there would be the best. Maybe we can use the git repo to store the data.tar.gz/xz/bz2/7z 

Compressed are the ~2.4gb only ~74mb big.

I'm currently not sure about the report_formats in this repo, if the path for them also wrong or correct. 

Maybe it would a cool idea to build the Images in 2 Versions one without initialized Data and one with.

Also don't "fuck up" with the symlinks. Or do it in a better way as current.

Currently this container has no quality that you should use in production.

Sorry to write it all here that not all match with the origin post.

Best regards Josef

**Comment by austinsonger** [comment]
@Dexus Yeah I agree with the switching to alpine, because it would help achieve this [milestone ](https://github.com/Secure-Compliance-Solutions-LLC/GVM-Docker/milestone/1).
